File: benchmarks/cli/csv_100k.py
import time, csv, io
data = '\n'.join(f'{i},name_{i},{i*3.14},{"active" if i%2==0 else "inactive"}' for i in range(100000))
start = time.monotonic_ns()
reader = csv.reader(io.StringIO(data))
rows = list(reader)
end = time.monotonic_ns()
print((end - start) // 1000)

File: benchmarks/cli/csv_100k.rb
require 'csv'
data = Array.new(100000) { |i| "#{i},name_#{i},#{(i*3.14).round(2)},#{i.even? ? 'active' : 'inactive'}" }.join("\n")
start = Process.clock_gettime(Process::CLOCK_MONOTONIC, :microsecond)
rows = CSV.parse(data)
finish = Process.clock_gettime(Process::CLOCK_MONOTONIC, :microsecond)
puts(finish - start)
